GitUI/CommandsDialogs/RevisionDiffController.cs

```diff
@@ -74,6 +78,96 @@ public sealed class ContextMenuSelectionInfo
 
     public sealed class RevisionDiffController : IRevisionDiffController
     {
+        private readonly Func<IGitModule> _getModule;
+        private readonly IFullPathResolver _fullPathResolver;
+
+        public RevisionDiffController(Func<IGitModule> getModule, IFullPathResolver fullPathResolver)
+        {
+            _getModule = getModule;
+            _fullPathResolver = fullPathResolver;
+        }
+
+        public void SaveFiles(List<FileStatusItem> files, Func<string, string?> userSelection)
+        {
+            if (files is null)
+            {
+                throw new ArgumentNullException(nameof(files));
+            }
+
+            if (files.Count == 0)
```

Comment:
❔ turn into `switch (files.Count)`

Reply:
I'm not seeing a good way of expressing all cases in a single switch because of `userSelection is null` check.